File: python/24b_1154A_RestoringThreeNumbers.py
def main():
    numbs = input().split()
    for i in range(4):
        numbs[i] = int(numbs[i])
    numbs.sort()
    a, b, c = numbs[3] - numbs[0], numbs[3] - numbs[1], numbs[3] - numbs[2]
    print(f"{a} {b} {c}")

if __name__ == "__main__":
    main()

File: cpp/24b_1154A_RestoringThreeNumbers.cpp
#include<bits/stdc++.h>

int main(){
    std::vector<int> numbs;
    int temp, a, b, c;
    for (int i = 0; i < 4; i++){
        std::cin>>temp;
        numbs.push_back(temp);
    }
    sort(numbs.begin(), numbs.end());
    a = numbs[3] - numbs[0];
    b = numbs[3] - numbs[1];
    c = numbs[3] - numbs[2];
    std::cout<<a<<" "<<b<<" "<<c;
    return 0;
}